src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -0,0 +1,27 @@
+import type { Token } from '$lib/types/token';
+import { nonNullish } from '@dfinity/utils';
+
+export const pinTokensAtTop = ({
+	$tokens,
+	$tokensToPin
+}: {
+	$tokens: Token[];
+	$tokensToPin: Token[];
+}): Token[] => {
+	const pinnedTokens = $tokensToPin
```

Comment:
For performance reason, can't we do this in a single shot / iteration? 

The `pinnedTokens` are equals to the `tokensToPin` no?

Reply:
Yes and no, in the sense that the `Token` object coming from `tokens` is for sure more complete than the ones in `tokensToPin`. It is not guaranteed that the object is exactly the same in the two lists.

Theoretically, `tokensToPin` could be even a simple list of `id` and `networkId`. So, the function `pinTokensAtTop` should receive a list to re-order and the way to order it.

I thought of doing like that. Do you prefer that the list `tokensToPin` has already the complete `Token` objects?

Reply:
> Yes and no, in the sense that the Token object coming from tokens is for sure more complete than the ones in tokensToPin.

Why? Which fields?

Typescript wise they are the same so if there are actually not we should be careful and more descriptive. You implementation then make sense but, we should solve this.

On the contrary, if they are the same, then I would try to improve the perf.

Reply:
understood; indeed the list tokensToPin should be better. What if:

```typescript
export type TokenToPin = Pick<Token, 'id'> & { network: Pick<Network, 'id'> };


const tokensToPin: Readable<TokenToPin[]> = derived(
	[icrcChainFusionDefaultTokens],
...


export const pinTokensAtTop = ({
	$tokens,
	$tokensToPin
}: {
	$tokens: Token[];
	$tokensToPin: TokenToPin[];
}): Token[] => {
...
```

Reply:
Yeah sounds cool!

In this PR or another PR? What do you rather like?

Reply:
better in this PR I think